tests/FSharp.Data.JsonValidation.Tests/Tests.fs

```diff
@@ -142,6 +142,15 @@ let ``a non-negative JSON number IsNonNegative`` () =
   invalid <| validate (NumberThat [IsNonNegative]) (JsonValue.Number -42M)
 
 [<Test>]
+let ``an integral JSON number IsIntegral`` () =
+  valid <| validate (NumberThat [IsIntegral]) (JsonValue.Number 42M) 
+  valid <| validate (NumberThat [IsIntegral]) (JsonValue.Number 0M)
+  valid <| validate (NumberThat [IsIntegral]) (JsonValue.Number -42M)
+  invalid <| validate (NumberThat [IsIntegral]) (JsonValue.Number 42.94M)
+  invalid <| validate (NumberThat [IsIntegral]) (JsonValue.Number -42.94M)
+  valid <| validate (NumberThat [IsIntegral]) (JsonValue.Float 0.0)
```

Comment:
It is interesting to note that a `JsonValue.Number` or `JsonValue.Float` ending with a `.0` will validate. 

I believe that this is reasonable because `42 = 42.0` mathematically speaking. I would love to hear your thoughts @mjgpy3.

Reply:
I think that's totally acceptable!